### DISCUSSION TITLE
Should there only be support for the latest version of DataFlex?

### DISCUSSION THREAD

**Discussion by NilsSve** [NilsSve/DfRefactor#50] [Polls]
Should there only be support for the latest DataFlex version when it comes to compiling?

Right now, DataFlex 19.0 and up are supported.
But as this is a developer tool, who doesn't have access to the latest DF?

**Comment by NilsSve** [comment]
I'm leaning towards only supporting the latest DF version...

**Comment by 2WA** [comment]
Make sense. Because the DFRefactor will assist the DataFlex Developers in migrating safely to the last version 23.0.
The developer is encouraged to migrate to use the new features like Unicode, FlexTron, new generation of connectivities, etc.

**Comment by sampizarro** [comment]
I agree in dropping oldest releases support. 

If you want to "keep" it somehow,  you can create a branch and tag it  with the corresponding DF release being dropped,  and keep it that way..  never merge them again..   Not an git expert, maybe some better options may exist..

**Reply by NilsSve** [reply]
I stink at Git, so if anybody knows how to do this and is willing to help, it would be a good thing to do something like this.

**Comment by mwpmullan** [comment]
There are reasons to upgrade only as far as 19.0, because unicode breaks other stuff, BUT we should encourage ourselves to just bite the damn bullet and move to the current version.  

/MM

**Reply by sampizarro** [reply]
yes,  that's why I proposed to create a branch, and freeze it..  
if someone needs , they can always clone from the branch corresponding tot he DF release they need..

**Reply by NilsSve** [reply]
Unicode is for sure a big challenge and this project is no exception. However, it should not affect any of the developer's projects run through the Refactoring engine as it is "neutral". Meaning that the engine keeps the file format untouched. So if a file is in UTF8 format (BOM) it is treated as UTF8 and if OEM, it is treated as OEM.

So there shouldn't be any reason why DFRefactor run under DataFlex 2023 (or later) couldn't be used for any older project (well, with the exception that there needs to be a *.sws or *.ws file)

So the plan is to only support for DataFlex 2023 (or later) and x64. That would mean far less code to maintain. As it is now all the libraries have both an x86 and an x64-bit folder and it would be nice to get rid of those not needed. And also all of the "#IF (!@ xxx" statements.

**Reply by sampizarro** [reply]
That all makes sense..  we just need to name our new branch properly, so we and end-user can easily identify them , if they are looking for DFRefactor release that supports 19.x  df versions.   I understand the plan is that in the next main branch updates,  we will be deleting the 19.x sws files,  and 32bit folders as you said..   

Any suggestion in the name ?   Also,  I can see currently 3 other branches in the repo.  It seems they are all merged,   do you need to keep them ?

**Reply by NilsSve** [reply]
Nope, there is no need to keep those repo's. In fact, it would be good to have them removed. But I don't know how(!)

**Reply by NilsSve** [reply]
For the branch name, perhaps "DFRefactor x64"?

**Reply by sampizarro** [reply]
no.. the new branch will not be "touched" anymore..  it's going to reflect the current state..  so it will represent DFRefactor releases that still support 19.x and x86 (32 bit) versions.        

the main branch will be the one that is going to be changed to only support x64 and DF 23.0 

So, maybe  "DFRefactor for DF 19.x (32bits/OEM)"

**Reply by NilsSve** [reply]
Oops, sorry. I misunderstood. Maybe without the /OEM bit so: "DFRefactor for DF 19 (32bit)"
If someone with more Git knowledge than me would like to do this, that would be great! (or instruct me how to do it)

**Reply by sampizarro** [reply]
I can do it..  and will also remove the older un-used branches..

**Reply by NilsSve** [reply]
Great Samuel, Please go ahead and do it!

**Reply by sampizarro** [reply]
Done @NilsSve .   new branch and tag/release created. old branches deleted! 

https://github.com/NilsSve/DfRefactor/releases

**Comment by NilsSve** [comment]
The voting is now closed.

The decision was to only support DataFlex 2023. Each new release will follow the latest DataFlex version.
